birb_registry/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+#![feature(const_trait_impl)]
+#![feature(fs_try_exists)]
+
+use birb::MainThreadModule;
+use std::{collections::HashMap, io::prelude::*, fs::File, fs};
+use std::fmt::{Debug, Formatter};
+use serde::{Serialize, Deserialize};
+
+#[derive(Default)]
+pub struct Registry {
+    ticks: u16,
```

Comment:
in my opinion, the Registry shouldn't be responsible for keeping track of time, or autosave intervals either, that should be up to the user, just provide a save function that takes a path and dumps to a file and maybe provide a timed autosave module as well that stores an interval. Maybe wait until the clock module gets merged